server/auth/providers/gitea.js

```diff
@@ -0,0 +1,130 @@
+// @flow
+import passport from "@outlinewiki/koa-passport";
+import Router from "koa-router";
+//import { capitalize } from "lodash";
+import { Strategy as GiteaStrategy } from "passport-gitea";
+import accountProvisioner from "../../commands/accountProvisioner";
+import env from "../../env";
+import {
+  GiteaOrganizationError,
+  GiteaOrganizationAPIError,
+} from "../../errors";
+import passportMiddleware from "../../middlewares/passport";
+//import { getAllowedDomains } from "../../utils/authentication";
+import { StateStore } from "../../utils/passport";
+
+const router = new Router();
+const providerName = "gitea";
+const GITEA_CLIENT_ID = process.env.GITEA_CLIENT_ID;
+const GITEA_CLIENT_SECRET = process.env.GITEA_CLIENT_SECRET;
+const GITEA_URL = process.env.GITEA_URL; //Usage of custom url
+//const allowedDomains = getAllowedDomains();
+
+const scopes = [];
+
+export const config = {
+  name: "Gitea",
+  enabled: !!GITEA_CLIENT_ID,
+};
+
+export async function request(endpoint: string, accessToken: string) {
+  const response = await fetch(endpoint, {
+    method: "GET",
+    headers: {
+      Authorization: `Bearer ${accessToken}`,
+      "Content-Type": "application/json",
+    },
+  });
+  return response.json();
+}
+
+if (GITEA_CLIENT_ID) {
+  let strategyOptions = {
+    clientID: GITEA_CLIENT_ID,
+    clientSecret: GITEA_CLIENT_SECRET,
+    authorizationURL: "",
+    tokenURL: "",
+    userProfileURL: "",
+    passReqToCallback: true,
+    callbackURL: `${env.URL}/auth/gitea.callback`,
+    store: new StateStore(),
+    scope: scopes,
+  };
+
+  //When using a user installed Gitea instance
+  if (GITEA_URL) {
```

Comment:
Because of the call on L72 to the api it seems like `GITEA_URL` is required, so we might as well just pass these in allways and as commented above just make sure it defaults to the hosted version?